### PR TITLE
lowell fixes

### PR DESCRIPTION
**Pull request by zpeyton**
Fixes issues with parsing the returned html

### CODE REVIEW

**Comment by iann** [comment]
Ran at ~7:00 PM and showing availability!
```
        {
            "name": "Lowell General",
            "hasAvailability": true,
            "availability": {
                "3/03/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 2
                },
                "3/05/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 10
                },
                "3/06/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 2
                },
                "3/07/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 2
                },
                "3/09/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 2
                },
                "3/10/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 3
                },
                "3/12/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 9
                },
                "3/13/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 3
                },
                "3/14/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 2
                },
                "3/16/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 1
                },
                "3/17/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 13
                },
                "3/19/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 32
                },
                "3/20/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 47
                },
                "3/21/2021": {
                    "hasAvailability": true,
                    "numberAvailableAppointments": 39
                }
            },
            "signUpLink": "https://www.lowellgeneralvaccine.com/#schedule",
            "startUrl": "https://www.lowellgeneralvaccine.com/schedule.html?wpforms%5Bfields%5D%5B2%5D=Massachusetts+residents+over+the+age+of+75+or+those+75%2B+with+a+Circle+Health+affiliated+PCP&wpforms%5Bfields%5D%5B5%5D%5B%5D=I+hereby+attest+under+penalty+of+perjury+and+to+the+best+of+my+knowledge+and+belief+that+I+%28or+the+individual+I+am+scheduling%29+belong+to+one+of+the+Massachusetts+Vaccination+Priority+Group+selected+above&wpforms%5Bfields%5D%5B6%5D%5B%5D=I+hereby+attest+under+penalty+of+perjury+that+I%2Fthey+reside%2C+work%2C+and%2For+study+in+the+State+of+Massachusetts&wpforms%5Bfields%5D%5B7%5D%5B%5D=I+understand+that+at+the+vaccination+site+I+may+be+asked+to+provide+proof+of+the+above.+Such+proof+can+include%3A+government+issued+ID%2C+work+badge%2C+employment+letter+or+current+pay+stub&wpforms%5Bid%5D=6400&wpforms%5Bauthor%5D=1&wpforms%5Bpost_id%5D=6037&wpforms%5Btoken%5D=dca7ffc8c0e098ddf465f4dc68841c1d",
            "address": "Cross River Center - East 1001 Pawtucket Blvd",
            "city": "Lowell",
            "state": "MA",
            "zip": "01854",
            "timestamp": "2021-03-02T00:00:37.782Z",
            "latitude": 42.6478169,
            "longitude": -71.342055
        },
```

**Review by iann — APPROVED** [review]
looks good

File: site-scrapers/LowellGeneral/index.js [hunk 1]
```diff
@@ -205,6 +234,14 @@ async function ScrapeWebsiteData(browser) {
...
                 stationAvailability[dateKey];
         }
+
+        // Sort the keys since we get dates out of order.
+        availability = Object.keys(availability)
+            .sort()
+            .reduce((obj, key) => {
+                obj[key] = availability[key];
+                return obj;
+            }, {});
```
Comment (johnhawkinson) [inline site-scrapers/LowellGeneral/index.js:244]: For what it's worth, lexicographical sorting is not correct for "m/dd/yyyy" dates. 
It won't be a real issue until October, I guess, but October sorts ahead of September.

```
> [ '1/1/2021' ,'9/1/2021', '10/1/2021' ].sort()
[ '1/1/2021', '10/1/2021', '9/1/2021' ]
```

Nothing appears to be zero-filling the months that I can tell (if I'm wrong, sorry!)

(I also dislike the .keys.sort.reduce idiom, but I don't have anything better to offer. ]

Comment (iann) [reply site-scrapers/LowellGeneral/index.js:244]: Good catch, you could use moment.diff as a custom sorter. Something like.
```
Object.keys(availability).sort((a, b) => {
  return moment(a).diff(b);
});
```

**Review by livgust — APPROVED** [review]
approving due to @iann 's review.

File: site-scrapers/LowellGeneral/index.js (hunk 1 above)
Comment (zpeyton) [reply site-scrapers/LowellGeneral/index.js:244]: Good call. I meant to put a lead zero on there like I do the day of month but forgot. I’ll put that in a new PR. Maybe we should validate the date key format?

**Pull request merged**